web_widget_slickroom/static/src/js/web_widget_slickroom.js

```diff
@@ -0,0 +1,65 @@
+/* Copyright 2017 LasLabs Inc.
+ * License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html). */
+
+odoo.define('web_widget_slickroom', function (require) {
+    "use strict";
+
+    var core = require('web.core');
+    var FieldSlickImages = require('web_widget_slick').FieldSlickImages;
+
+    var FieldSlickroomImages = FieldSlickImages.extend({
+
+        widget_class: FieldSlickImages.prototype.widget_class + ' o_slickroom',
+        events: _.extend({}, FieldSlickImages.prototype.events, {
+            'click img': '_openModal'
+        }),
+
+        _openModal: function (ev) {
+            if (this.get("effective_readonly")) {
+                return;
+            }
+
+            var recordId = Number($(ev.target).attr('data-record-id'));
+            var modalAction = {
+                type: 'ir.actions.act_window',
+                res_model: 'darkroom.modal',
+                name: 'Darkroom',
+                views: [[false, 'form']],
+                target: 'new',
+                context: {
+                    active_field: this.options.fieldName,
+                    active_model: this.options.modelName,
+                    active_record_id: recordId
+                }
+            };
+
+            this.do_action(modalAction, {
+                on_close: $.proxy(this._updateImage, this, [recordId])
+            });
+        },
+
+        _slickRender: function(baseUrl, id) {
+            this._super(baseUrl, id);
+            this.$slick.find('img').last().attr('data-record-id', id);
+        },
+
+        _updateImage: function (recordId) {
+            // SlickJS creates 'clones', so multiple elements need updated src
+            var $imgs = this.$slick.find('[data-record-id="' + recordId + '"]');
```

Comment:
`img[data-record-id="' + recordId + '"]` would be better IMO - slightly more explicit for no performance overhead

Reply:
I had to change this line as a result of using jquery `.data()` vs `.attr()` per your other comments, but I did add the `img` selector to be more explicit.